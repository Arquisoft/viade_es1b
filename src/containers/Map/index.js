import Mapa from './map.component';



export default Mapa;
