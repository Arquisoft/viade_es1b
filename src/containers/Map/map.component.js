import React from 'react';
import L from 'leaflet';
import { TileLayer, Marker, Polyline, Popup } from 'react-leaflet';
import Rutas from './routes/rutas';
import { LoggedOut, LoggedIn } from '@solid/react';
import { Redirect } from 'react-router-dom';
import ReactDOM from 'react-dom';
import { MapStyle, DivStyle } from './map.style';

delete L.Icon.Default.prototype._getIconUrl;

L.Icon.Default.mergeOptions({
  iconRetinaUrl: require('leaflet/dist/images/marker-icon-2x.png'),
  iconUrl: require('leaflet/dist/images/marker-icon.png'),
  shadowUrl: require('leaflet/dist/images/marker-shadow.png')
});

class Map extends React.Component {

  constructor() {
    super();
    this.name = Rutas.getNames()[0];
    this.puntos = []
    Rutas.getRutaByPosition(1).points.map(p => this.puntos.push(p.getCoordinates()));
  }

  getRoutes(id, e) {


    var newRuta = Rutas.getRutaByName(id);
    document.getElementById("name").textContent = newRuta.name;

    this.puntos = newRuta.point;
    const position = this.puntos[0];

    var update = <MapStyle id="map" center={position} zoom={15} >
      <TileLayer url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png" />
      <Polyline color={'red'} positions={this.puntos}></Polyline>
      <Marker position={this.puntos[0]}>
        <Popup>Inicio</Popup>
      </Marker>
      <Marker position={this.puntos[this.puntos.length - 1]}>
        <Popup>Fin</Popup>
      </Marker>
    </MapStyle>;

    ReactDOM.render(update, document.getElementById('map'));
  }

  render() {
    Rutas.actualizarRutasConPod();
    const position = this.puntos[0];





    return (

      <React.Fragment id="map" >
        <LoggedIn>
          <DivStyle>
            <h2 id="name">{this.name}</h2>
            <ul>{Rutas.getNames().map((n, i) => <li key={i} onClick={(e) => this.getRoutes(n, e)}> {n} </li>)}</ul>
          </DivStyle>
          <MapStyle id="map" center={position} zoom={15}>
            <TileLayer url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png" />
            <Polyline color={'red'} positions={this.puntos}></Polyline>
            <Marker position={this.puntos[0]}>
              <Popup>Inicio</Popup>
            </Marker>
            <Marker position={this.puntos[this.puntos.length - 1]}>
              <Popup>Fin</Popup>
            </Marker>
          </MapStyle>
        </LoggedIn>
        <LoggedOut>
          <Redirect to='/login'></Redirect>
        </LoggedOut>
      </React.Fragment>

    );
  }
}

export default Map;
