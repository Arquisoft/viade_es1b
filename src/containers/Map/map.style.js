import styled from 'styled-components';
import { Map } from "react-leaflet";
import 'leaflet/dist/leaflet.css';

export const MapStyle = styled(Map)`
    position: absolute;
    width: 100%;
    height: 100%;
    z-Index: 1;
  `;

export const DivStyle = styled.div`
    position: absolute;
    background-image: linear-gradient( white, lightgrey);
    border: 1px solid lightgray;
    padding: 20px;
    width: 30%;
    height: auto;
    max-height: 50%;
    z-Index: 99;
    overflow: auto;
    align:left;
  `;

export const DivStyle3 = styled.div`
  overflow: auto; 
`;

export const DivStyle4 = styled.div`
  background-image: linear-gradient( white, white);
  border: 1px solid gray;
  overflow: auto;
  button{
    width:100%;
    text-align: left;
  }
`;

export const AmigosDiv = styled.div`
   border: 1px solid gray;
   padding: 5px;
`;

export const LiStyle = styled.li`
    list-style: none;
    width : 100%;
    border: 1px solid gray;
    background-color: white;
    text-align:left;
    padding: 5px;
  `;

export const H3Style = styled.h3`
    text-align:center;
    font-family: Arial, Helvetica, sans-serif;
  `;

export const LiStyle2 = styled.div`
  border: 1px solid gray;
  list-style: none;
  width : 95%;
  text-align:center;
  padding: 5px;
  margin: 10px;
`;

export const InputStyle = styled.input`
    width: auto;
    height: auto;
    `;




