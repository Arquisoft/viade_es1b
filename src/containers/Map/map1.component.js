/* eslint-disable constructor-super */
/* eslint-disable no-nested-ternary */
/* eslint-disable no-console */

/*import React from 'react'
import { Map as LeafletMap, TileLayer, Marker, Popup } from 'react-leaflet';
import L from 'leaflet';
import './leaflet.css';
import {
  MapContainer,
  H2prueba,
  Derecha,
  MapSection,
  Button,
  MapSize
} from './map.style';

const HtmlLeaflet = ({ webId }: Props) => {
  console.log(webId);
  return (
    <html>
        <meta http-equiv="content-type" content="text/html; charset=UTF-8"></meta>
        <link rel="stylesheet" href="Using%20GeoJSON%20with%20Leaflet%20-%20Leaflet%20-%20a%20JavaScript%20library%20for%20interactive%20maps_files/leaflet.css" integrity="sha512-xwE/Az9zrjBIphAcBb3F6JVqxf46+CDLwfLMHloNu6KEQCAWi6HcDUbeOfBIptF7tcCzusKFjFw2yuvEpDL9wQ==" crossorigin=""></link>
        <script src="Using%20GeoJSON%20with%20Leaflet%20-%20Leaflet%20-%20a%20JavaScript%20library%20for%20interactive%20maps_files/leaflet.js" integrity="sha512-gZwIG9x3wUXg2hdXF6+rVkLF/0Vi9U8D2Ntg4Ga5I5BZpVkVxlJWbSQtXPSiUTtC0TjtGOmxa1AJPuV0CPthew==" crossorigin=""></script>
    </html>
   
  );
};


const position = [51.505, -0.09]

const Map = ({ webId }: Props) => {
  console.log(webId);
  return (
    <MapSize>
      <LeafletMap center={position} zoom={13}>
        <TileLayer
        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        attribution="&copy; <a href=&quot;http://osm.org/copyright&quot;>OpenStreetMap</a> contributors"
        />
        
      </LeafletMap>
    </MapSize>
    

 
  );
};



/*class Map extends React.Component {
  render() {
    return (
      <MapSize>
        <LeafletMap
          center={[50, 10]}
          zoom={6}
          maxZoom={10}
          attributionControl={true}
          zoomControl={true}
          doubleClickZoom={true}
          scrollWheelZoom={true}
          dragging={true}
          animate={true}
          easeLinearity={0.35}
        >
          <TileLayer
            url='http://{s}.tile.osm.org/{z}/{x}/{y}.png'
          />
          <Marker position={[50, 10]}>
            <Popup>
              Popup for any custom information.
            </Popup>
          </Marker>
        </LeafletMap>
      </MapSize>;
    );
  }
}

const styles = {
  wrapper: { 
    height: 400, 
    width: '80%', 
    margin: '0 auto', 
    display: 'flex' 
  },
  map: {
    flex: 1
  } 
};

class Mapa extends React.Component {
  render() {
    return (
        <MapSection>
          <MapContainer>
            <Moves></Moves>
          </MapContainer>
        </MapSection>
  
    );
  }
}

export default Mapa
*/


import { Map, TileLayer } from 'react-leaflet';
import L from 'leaflet';