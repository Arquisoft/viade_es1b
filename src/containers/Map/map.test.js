import React from 'react';
import { render, queryByTestId, queryById } from '@testing-library/react';
import Map from './map.component';

let wrapper;
beforeEach(() => {
  const { container, debug } = render(
    <Map />
  );
  wrapper = container;
  //debug();
});

describe('Map Page Render', () => {


  test('renders without crashing', () => {
    expect(wrapper).toBeTruthy();
  });

  test('renders all components', () => {
    expect(queryByTestId(wrapper, "map-title")).not.toBeNull();
    expect(queryByTestId(wrapper, "map-routes-list")).not.toBeNull();
    expect(wrapper.querySelector('#map')).not.toBeNull();
  });


});

