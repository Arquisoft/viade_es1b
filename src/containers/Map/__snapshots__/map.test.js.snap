// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Page Render renders with default props 1`] = `ShallowWrapper {}`;
